test-fork/ChefSiphon.spec.ts

```diff
@@ -0,0 +1,184 @@
+        let chefForwarderSiphonToken: SiphonToken;
+
+        // -------------------------------------------------------------------
+        // EOA will perform these actions
+        // -------------------------------------------------------------------
+
+        const mintAmount = parseUnits("1");
+
+        it("deploy ChefForwarder", async () => {
+            chefForwarder = await deployContract<ChefForwarder>(hre, new ChefForwarder__factory(eoa), "ChefForwarder", [
+                masterChef.address,
+            ]);
+            await chefForwarder.setBriber(eoaAddress);
+        });
+        it("deploy siphon tokens for chefForwarder", async () => {
+            chefForwarderSiphonToken = await deployContract<SiphonToken>(
+                hre,
+                new SiphonToken__factory(eoa),
+                "SiphonTokenBribes",
+                [chefForwarder.address, mintAmount],
+                {},
+                debug,
+            );
+            const balance = await chefForwarderSiphonToken.balanceOf(chefForwarder.address);
+            expect(balance).eq(mintAmount);
+        });
+        it("deploy MasterChefRewardHook", async () => {
+            masterChefRewardHook = await deployContract<MasterChefRewardHook>(
+                hre,
+                new MasterChefRewardHook__factory(eoa),
+                "MasterChefRewardHook",
+                [stashAddress, masterChef.address, auraToken.address],
+            );
+        });
+        it("deploy siphon tokens for masterChefRewardHook", async () => {
+            masterChefRewardHookSiphonToken = await deployContract<SiphonToken>(
+                hre,
+                new SiphonToken__factory(eoa),
+                "SiphonTokenBribes",
+                [masterChefRewardHook.address, mintAmount],
+                {},
+                debug,
+            );
+            const balance = await masterChefRewardHookSiphonToken.balanceOf(masterChefRewardHook.address);
+            expect(balance).eq(mintAmount);
+        });
+        it("transfer ownership of reward contracts to protocolMultisig", async () => {
+            await masterChefRewardHook.transferOwnership(protocolMultisig);
+            await chefForwarder.transferOwnership(protocolMultisig);
+
+            masterChefRewardHook = masterChefRewardHook.connect(protocolDao);
+            chefForwarder = chefForwarder.connect(protocolDao);
+        });
+
+        // -------------------------------------------------------------------
+        // Protocol DAO will perform these actions
+        // -------------------------------------------------------------------
+        const poolAllocPoint = "1000";
+        const rewarder = zeroAddress;
+
+        it("add siphon token for masterChefRewardHook", async () => {
+            const pid = await masterChef.poolLength();
+            await masterChefRewardHook.setPid(pid);
+            await masterChef.add(poolAllocPoint, masterChefRewardHookSiphonToken.address, rewarder);
```

Comment:
Would default to just using `ZERO_ADDRESS` rather than setting `rewarder = zeroAddress` - easier to parse